src/bin/evtx_dump.rs

```diff
@@ -225,45 +204,41 @@ impl EvtxDump {
                         Ok(File::create(p)?)
                     }
                 }
-                None => err!("Output file cannot be root."),
+                None => bail!("Output file cannot be root."),
             }
         }
     }
 
-    fn dump_record(
-        &mut self,
-        record: Result<SerializedEvtxRecord<String>, Error>,
-    ) -> Result<(), Error> {
-        match record {
+    fn dump_record(&mut self, record: EvtxResult<SerializedEvtxRecord<String>>) -> Result<()> {
+        match record.with_context(|| "Failed to dump the next record.") {
             Ok(r) => {
                 if self.show_record_number {
                     writeln!(self.output, "Record {}", r.event_record_id)?;
                 }
                 writeln!(self.output, "{}", r.data)?;
             }
+            // This error is non fatal.
             Err(e) => {
-                if self.backtraces {
-                    dump_err_with_backtrace(&e)
-                } else {
-                    eprintln!("{:?}", &e);
+                eprintln!("{:?}", format_err!(e));
+
+                if self.stop_after_error {
+                    std::process::exit(1);
                 }
             }
         };
 
         Ok(())
     }
 
-    fn try_to_initialize_logging(&self) {
-        if let Some(level) = self.verbosity_level {
-            match simplelog::WriteLogger::init(
+    fn try_to_initialize_logging(&self) -> Result<()> {
+        Ok(if let Some(level) = self.verbosity_level {
```

Comment:
Wow wired

Maybe use the explicit `Ok(())`?

Reply:
👍